packages/vitest-pool-workers/src/worker/fetch-mock.ts

```diff
@@ -21,17 +21,19 @@ interface BufferedRequest {
 	request: Request;
 	body: Uint8Array | null;
 }
-const requests = new WeakMap<Dispatcher.DispatchOptions, BufferedRequest>();
-const responses = new WeakMap<Dispatcher.DispatchOptions, Response>();
+const requests = new Map<string, BufferedRequest>();
+const responses = new Map<string, Response>();
```

Comment:
In an ideal world we would wrap these in a class that made sure items were removed when we get them to avoid having to remember doing so at each callsite.